src/lib/config.rs

```diff
@@ -229,86 +229,58 @@ impl Config {
   /// Convenience function that allows getting a HashMap from a `Config` of
   /// the `ManagedObject`s within.
   pub fn as_managed_objects(config: Config) -> HashMap<String, ManagedObject> {
-    let mut mos: HashMap<String, ManagedObject> = HashMap::new();
-    for _f in config.files.iter() {
-      let mut mo = ManagedObject::default();
-      mo.name = _f.0.to_owned();
-      match _f.1.get("solution") {
-        None => (),
-        Some(_x) => {
+    config
+      .files
+      .iter()
+      .map(|(name, val)| {
+        let mut mo = ManagedObject::default();
+        mo.name = name.to_owned();
+        if let Some(_x) = val.get("solution") {
           mo.solution = String::from(_x.as_str().unwrap());
         }
-      }
-      match _f.1.get("task") {
-        None => (),
-        Some(_x) => {
+        if let Some(_x) = val.get("task") {
           mo.task = String::from(_x.as_str().unwrap());
         }
-      }
-      match _f.1.get("source") {
-        None => (),
-        Some(_x) => {
+        if let Some(_x) = val.get("source") {
           mo.source = String::from(_x.as_str().unwrap());
         }
-      }
-      match _f.1.get("method") {
-        None => (),
-        Some(_x) => {
+        if let Some(_x) = val.get("method") {
           mo.method = String::from(_x.as_str().unwrap());
         }
-      }
-      match _f.1.get("destination") {
-        None => (),
-        Some(_x) => {
+        if let Some(_x) = val.get("destination") {
           mo.destination = String::from(_x.as_str().unwrap());
         }
-      }
-      match _f.1.get("dependencies") {
-        None => (),
-        Some(_x) => {
-          let _f = _x.as_str().unwrap();
-          // thanks https://stackoverflow.com/a/37547426
-          mo.dependencies = _f.split(", ").map(|s| s.to_string()).collect();
+        if let Some(_x) = val.get("dependencies") {
+          let _f = _x.as_array().unwrap();
+          mo.dependencies = _f.iter().map(|v| v.as_str().unwrap().to_owned()).collect();
         }
-      }
-      match _f.1.get("force") {
-        None => (),
-        Some(_x) => {
-          let _f = _x.as_str().unwrap();
+        if let Some(_x) = val.get("force") {
           // haha boolean assignment go brr
-          mo.force = _f == "true";
+          mo.force = _x.as_bool().unwrap();
```

Comment:
This actually crashes with a `panic` in my `config.toml`, as my `'force'` entries are single quoted, and the parsing works only on the bare string `force`. It may be worth massaging this about a bit to get either to be valid?